### PR TITLE
feat: Implement `.fetchOneBuild()` in the GithubActionsSourceClientAdapter

### LINKED ISSUE #1287: CI Integrations: GithubActionsSourceClientAdapter - implement `.fetchOneBuild` method
**Description**
Implement `.fetchOneBuild` method for the BuildkiteSourceClientAdapter to match the updated interface.

**Acceptance Criteria**
- [ ]  `.fetchOneBuild()` method implemented.
- [ ] `.fetchOneBuild()` tested .

### PR DESCRIPTION
**Pull request by solid-vovabeloded**
Fixes #1287 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart [hunk 1]
```diff
@@ -1172,11 +1173,833 @@ void main() {
...
+        await adapter.fetchOneBuild(jobName, buildNumber);
+
+        verifyNever(
+          githubActionsClientMock.fetchRunJobsNext(workflowRunJobsPage),
+        );
+      },
+    );
+
+    test(
+      ".fetchOneBuild() does not fetch the next workflow run jobs page if the previous one contains the job with the given name",
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/test/source/github_actions/adapter/github_actions_source_client_adapter_test.dart:1530]: The same suggestion here, as for workflow runs page in the above.

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Good job, @solid-vovabeloded! I've added a couple of suggestions below - please let me know what you think! Overall, the changes look great and very well-tested - thank you!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-vovabeloded!:rocket:

**Comment by illia-romanenko** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**